common/source/serial_port.cpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * serial_port.cpp
+ *
+ *  Created on: 2020��9��18��
+ *      Author: duanning
+ */
+
+#include "serial_port.h"
+
+SerialPort::SerialPort(const string& name):
+	serial_port_(io_s_) {
+	CommonKeyPairs config = {
+			{"hid_name", name},
+	};
+
+	ReConfig(config);
+}
+
+SerialPort::SerialPort(const string& name, AbstractProtocol& protocol):
+	serial_port_(io_s_),
+	application_protocol_(protocol) {
```

Comment:
`application_protocol_` is initialized but never referenced. IMHO the protocol is used when `Set` and `Get` data. At least we should pack or unpack the packet.